docs/fhir-guide/FHIR 101 - Practical Guide.ipynb

```diff
@@ -0,0 +1,2966 @@
+     ]
+    }
+   ],
+   "source": [
+    "# Let's create an invalid resource and make sure it fails \n",
+    "# Invalid since its missing the REQUIRED gender attribute\n",
+    "patient_rs_invalid = {\n",
+    "    \"resourceType\":\"Patient\",\n",
+    "    \"id\": \"kidsfirst-test-patient\",\n",
+    "    \"text\": {\n",
+    "        \"status\": \"generated\",\n",
+    "        \"div\": \"<div xmlns=\\\"http://www.w3.org/1999/xhtml\\\"><p>An invalid patient resource for testing purposes</p></div>\"\n",
+    "    },\n",
+    "    \"meta\": {\n",
+    "        \"profile\": [\"http://fhir.kids-first.io/StructureDefinition/Patient\"]\n",
+    "    }\n",
+    "}\n",
+    "validate_rs(patient_rs_invalid, expected_success=False, print_result=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true,
+    "hidden": true
+   },
+   "source": [
+    "### Add an Attribute"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "#### Extensions\n",
+    "In FHIR, when you want to add a new attribute to a `StructureDefinition`, its known as adding an \"extension\" since you are extending the definition of the base model.\n",
+    "\n",
+    "To add an attribute to the `Patient` `StructureDefinition`, you must do 2 things: \n",
+    "1. Define an extension for the attribute by creating a `StructureDefinition` resource for it\n",
+    "2. Declare/reference the extension in the `Patient` `StructureDefinition` \n",
+    "\n",
+    "📘Learn more about defining [extensions](https://www.hl7.org/fhir/extensibility-examples.html)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "#### Why so complicated? ----> Reusability\n",
+    "You might be wondering why adding a new attribute isn't as simple as removing an attribute. There are a couple of reasons for this. The primary reason is because the FHIR spec pushes for interoperability through sharing of model components like extensions.\n",
+    "\n",
+    "Since extensions are defined in their own files, they can be reused in a plug and play fashion by other models. We will see this later in the tutorial when we reuse the `karyotypic-sex`extension defined by the [Phenopackets FHIR model](https://aehrc.github.io/fhir-phenopackets-ig/).\n",
+    "\n",
+    "Unfortunately, if you don't care about making your attributes shareable and/or you only care about simple key-value type attributes, FHIR does not have a simpler way to define these right now.\n",
+    "\n",
+    "\n",
+    "#### Simple and Complex Extensions\n",
+    "Another reason why extensions may be complex is because they can represent both simple attributes or complex attributes. A simple attribute might be a key, value pair with a primitive data type and a complex attribute might be an entirely new structure that might have several other extensions in a nested hierarchy.\n",
```

Comment:
```suggestion
    "Another reason why extensions may be complex is because they can represent both simple attributes or complex attributes. A simple attribute might be a key:value pair with a primitive data type, and a complex attribute might be an entirely new structure that might have several other extensions in a nested hierarchy.\n",
```